test/utils/helpers.go

```diff
@@ -612,3 +624,21 @@ func WaitApiServer(k8sClient client.Client, timeout time.Duration) error {
 	}
 	return nil
 }
+
+// GetResourceNameFromAzureResourceID retrieves the VM name from Azure resource ID.
+func GetResourceNameFromAzureResourceID(resourceID string) (string, error) {
```

Comment:
Would it be better to add a VMName output in terraform to retrieve this information, instead of parsing the id string? If we decide to just parse the id string, consider to use `len(lowerCaseIDFields) - 1` instead of check for length and hardcode index 7, in case azure changes resource path?